### PR TITLE
allow te to use meta device with deferred init

### PR DESCRIPTION
**Pull request by cli99**
Specifying the `meta` device (`init_device: meta`) triggers the deferring init (defer module parameter initialization until after FSDP sharding)  in transformer engine repo. This resolves the excessive memory usage problem when using FSDP + te FP8.  The fix goes with the NV te fix https://github.com/NVIDIA/TransformerEngine/pull/596.

### CODE REVIEW

**Review by j316chuck — APPROVED** [review]
Great find! Curious how you found this fix?

**Comment by cli99** [comment]
@j316chuck yes, we identified the excessive memory problem and verified fix a while ago, with the help of the memory profiling.

**Review by mvpatel2000 — APPROVED** [review]

File: llmfoundry/models/layers/ffn.py [hunk 1]
```diff
@@ -97,8 +97,8 @@ def __init__(
         self.fc_kwargs: dict[str, Any] = {
             'bias': bias,
         }
-        if fc_type != 'te':
-            self.fc_kwargs['device'] = device
+
+        self.fc_kwargs['device'] = device
```
Comment (vchiley) [inline llmfoundry/models/layers/ffn.py:101]: y not just add at line 99

File: llmfoundry/models/layers/attention.py [hunk 2]
```diff
@@ -522,8 +522,7 @@ def __init__(
         fc_kwargs: dict[str, Any] = {
             'bias': bias,
         }
-        if fc_type != 'te':
-            fc_kwargs['device'] = device
+        fc_kwargs['device'] = device
```
Comment (vchiley) [inline llmfoundry/models/layers/attention.py:525]: y not just add at line 524

**Pull request merged**